internal/pipe/changelog/changelog.go

```diff
@@ -61,7 +79,19 @@ func (Pipe) Run(ctx *context.Context) error {
 		log.Debug("is gitlab or gitea changelog")
 		changelogStringJoiner = "   \n"
 	}
-	ctx.ReleaseNotes = fmt.Sprintf("## Changelog\n\n%v\n", strings.Join(entries, changelogStringJoiner))
+
+	notesBuilder := strings.Builder{}
+	notesBuilder.WriteString("## Changelog\n\n")
+	if ctx.ReleaseHeader != "" {
+		notesBuilder.WriteString(ctx.ReleaseHeader + "\n\n")
+	}
+	changeLog := strings.Join(entries, changelogStringJoiner)
+	notesBuilder.WriteString(changeLog)
+	if ctx.ReleaseFooter != "" {
+		notesBuilder.WriteString(ctx.ReleaseFooter + "\n\n")
+	}
+	notesBuilder.WriteString("\n")
+	ctx.ReleaseNotes = notesBuilder.String()
```

Comment:
maybe:

```go
ctx.ReleaseNotes = strings.Join(
  []string{
     ctx.ReleaseHeader,
     "## Changelog",
     strings.Join(entries, changelogStringJoiner),
     ctx.ReleaseFooter,
  },
  "\n\n",
)
```

it might add some extra empty lines, but AFAIK github handles it... if not, maybe:

if not, the way you did seems fine :)